### PR TITLE
Update EcoRRAP sites with new locations, and fix plot_map() function. 

### PR DESCRIPTION
**Pull request by eco-ben**
Changed EcoRRAP site locations to the new locations provided by EcoRRAP team. Converted `plot_map()` function into a single function definition with optional argument color_by with a default value for no colour plotting.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -230,7 +233,14 @@ vessels and divers) is fully utilised, starting by selecting randomly from the t
 then randomly from the priority reefs, and then - if there is spare capacity, which is
 rarely - randomly from the non-priority.
 
-### Depth Quality Control flags
+### Depth Data and Quality Control flags
+
+For calculation of pixel size depth raster data should have coordinates in unit metres, not
+degrees.
```
Comment (ConnectedSystems) [inline README.md]: Change this statement to be more general and move to the top, making clear that 

1. this set of scripts (attempts to) creates a standard set of geospatial data in a projection with a meter coordinate system
2. All functionality expects meter units

File: README.md [hunk 2]
```diff
@@ -239,9 +249,16 @@ The `depth_qc` attribute values indicate:
 - 0 : no error (does not indicate polygons that only partially overlapped a given reef!)
 - 1 : flags that the reef feature did not overlap any satellite data (value set to 7m)
 - 2 : flags that the minimum value was above sea level (no changes/adjustments made)
+- 3 : flags that depth raster data used for statistic calculation covers less than
+5% of the polygon reef area
```
Comment (ConnectedSystems) [inline README.md:254]: Looking at error code 1, maybe we should flag it with `3` but set the value to 7m as well?

File: README.md [hunk 3]
```diff
@@ -239,9 +249,16 @@ The `depth_qc` attribute values indicate:
...
 
 ### Coordinate Reference Systems
 
 `1_create_canonical.jl` uses a Great Barrier Reef Features dataset to create the initial
 canoical dataset. These features are in crs EPSG:4283 (GDA1994). These features are then
 reprojected to be in crs EPSG:7844 (GDA2020) to be consistent with other data from GBRMPA.
+
+### Possible error reef features
```
Comment (ConnectedSystems) [inline README.md]:
```suggestion
### Possible error in reef features
```

File: README.md [hunk 4]
```diff
@@ -239,9 +249,16 @@ The `depth_qc` attribute values indicate:
...
 - 1 : flags that the reef feature did not overlap any satellite data (value set to 7m)
 - 2 : flags that the minimum value was above sea level (no changes/adjustments made)
+- 3 : flags that depth raster data used for statistic calculation covers less than
+5% of the polygon reef area
 
 ### Coordinate Reference Systems
 
 `1_create_canonical.jl` uses a Great Barrier Reef Features dataset to create the initial
 canoical dataset. These features are in crs EPSG:4283 (GDA1994). These features are then
 reprojected to be in crs EPSG:7844 (GDA2020) to be consistent with other data from GBRMPA.
```
Comment (ConnectedSystems) [inline README.md]: Not part of this PR, but note CRS is an acronym, so should be capitalized.

File: src/10_extract_reef_depths.jl [hunk 5]
```diff
@@ -41,11 +42,19 @@ gbr_features = GDF.read(canonical_dataset)
...
+        return [-maximum(x), -mean(x), -median(x), -minimum(x), std(x), prop_area]
+    else
+        return [-first(x), -first(x), -first(x), -first(x), -first(x), prop_area]
+    end
+end
 
 # Using SharedArrays now as we might move to multi-core processing
 # although using GPUs is an option...
-depths = SharedArray(zeros(Float64, size(gbr_features, 1), 5))
+depths = SharedArray(zeros(Float64, size(gbr_features, 1), 6))
```
Comment (ConnectedSystems) [inline src/10_extract_reef_depths.jl:57]: Why was SharedArray needed here?

Comment (ConnectedSystems) [reply src/10_extract_reef_depths.jl:57]: Oh nevermind, read the comment above.

File: README.md [hunk 6]
```diff
@@ -24,6 +24,8 @@ Generates a standardized geopackage file including data from:
   - http://www.nntt.gov.au/assistance/Geospatial/Pages/DataDownload.aspx
 - Satellite-derived Bathymetry data at 10m resolution
   - https://gbrmpa.maps.arcgis.com/home/item.html?id=f644f02ec646496eb5d31ad4f9d0fc64
+- QLD Port Location data
```
Comment (ConnectedSystems) [inline README.md:29]: Where is this from?

File: README.md [hunk 7]
```diff
@@ -24,6 +24,8 @@ Generates a standardized geopackage file including data from:
...
+- EcoRRAP site locations from EcoRRAP team (Maren Toor)
```
Comment (ConnectedSystems) [inline README.md]: Dr. Toor I believe 😃

**Review by ConnectedSystems — CHANGES_REQUESTED** [review]
Minor changes in documentation, but otherwise looks good!

File: README.md (hunk 1 above)
Comment (eco-ben) [reply README.md]: I think the meter coordinate requirement only applies to the depth data. The output features etc should be in GDA2020 degrees.
